mu/logic.py

```diff
@@ -1204,11 +1205,12 @@ def load(self, *args, default_path=None):
         extracts a Python script from a hex file.
         """
         # Get all supported extensions from the different modes
-        extensions = ["py"]
+        extensions = [ext.strip("*.") for ext in self.python_extensions]
         for mode_name, mode in self.modes.items():
             if mode.file_extensions:
                 extensions += mode.file_extensions
-        extensions = set([e.lower() for e in extensions])
+        # Sort extensions to get deterministic output
+        extensions = list(sorted(set([e.lower() for e in extensions])))
```

Comment:
I guess you don't need set() here?

Reply:
As things stand, no, using a set() here doesn't help at all and introduces unstable ordering. It can be argued that it might help when adding modes, but IMO it'd need a test exercising extension deduplication to justify the set() staying.

Having the results sorted or ordered seems desirable, removing the set and sort gives us order. I'll try it like that.